consensus/view_change.go

```diff
@@ -533,14 +539,19 @@ func (consensus *Consensus) onNewView(msg *msg_pb.Message) {
 	// TODO: check magic number 32
 	if len(recvMsg.Payload) > 32 {
 		// Construct and send the commit message
-		blockNumHash := make([]byte, 8)
-		binary.LittleEndian.PutUint64(blockNumHash, consensus.blockNum)
+		commitPayload := signature.ConstructCommitPayload(consensus.ChainReader,
+			consensus.blockHash, consensus.epoch, consensus.blockNum, consensus.viewID)
```

Comment:
@gupadhyaya please double check the view change logic.

Reply:
I think we should have use the recvMsg's data for ViewID

Reply:
using `consensus.viewID` is ok here, as line 518 sets it `consensus.viewID = recvMsg.ViewID`